seqr/views/utils/terra_api_utils_tests.py

```diff
@@ -0,0 +1,90 @@
+                   '{"accessLevel": "READER","public": true,"workspace": {"attributes": {"tag:tags": {"itemsType": "AttributeValue","items": ["differential-expression","tutorial"]},"description": "[DEGenome](https://github.com/eweitz/degenome) transforms differential expression data into inputs for [exploratory genome analysis with Ideogram.js](https://eweitz.github.io/ideogram/differential-expression?annots-url=https://www.googleapis.com/storage/v1/b/degenome/o/GLDS-4_array_differential_expression_ideogram_annots.json).  \\n\\nTry the [Notebook tutorial](https://app.terra.bio/#workspaces/degenome/degenome/notebooks/launch/degenome-tutorial.ipynb), where you can step through using DEGenome to analyze expression for mice flown in space!"},"authorizationDomain": [],"bucketName": "fc-2706d493-5fce-4fb2-9993-457c30364a06","createdBy": "eric.m.weitz@gmail.com","createdDate": "2020-01-14T10:21:14.575Z","isLocked": false,"lastModified": "2020-02-01T13:28:27.309Z","name": "degenome","namespace": "degenome","workflowCollectionName": "2706d493-5fce-4fb2-9993-457c30364a06","workspaceId": "2706d493-5fce-4fb2-9993-457c30364a06"},"workspaceSubmissionStats": {"runningSubmissionsCount": 0}},'
+                   '{"accessLevel": "PROJECT_OWNER","public": false,"workspace": {"attributes": {"description": "A workspace for seqr project"},"authorizationDomain": [],"bucketName": "fc-6a048145-c134-4004-a009-42824f826ee8","createdBy": "zhangshifa07504@gmail.com","createdDate": "2020-09-09T15:12:30.142Z","isLocked": false,"lastModified": "2020-09-09T15:12:30.145Z","name": "seqr-project 1000 Genomes Demo","namespace": "my-seqr-billing","workflowCollectionName": "6a048145-c134-4004-a009-42824f826ee8","workspaceId": "6a048145-c134-4004-a009-42824f826ee8"},"workspaceSubmissionStats": {"runningSubmissionsCount": 0}}]')
+        workspaces = list_anvil_workspaces(self.user)
+        self.assertEqual(len(workspaces), 3)
+        self.assertIsNotNone(workspaces[0]['public'])
+
+        responses.replace(responses.GET, url, status = 200,
+            body = '[{"accessLevel": "PROJECT_OWNER", "workspace": {"name": "1000 Genomes Demo", "namespace": "my-seqr-billing", "workspaceId": "237998e6-663d-40b9-bd13-57c3bb6ac593" }},'
+                   '{"accessLevel": "READER","workspace": {"name": "degenome","namespace": "degenome", "workspaceId": "2706d493-5fce-4fb2-9993-457c30364a06"}},'
+                   '{"accessLevel": "PROJECT_OWNER","workspace": {"name": "seqr-project 1000 Genomes Demo","namespace": "my-seqr-billing","workspaceId": "6a048145-c134-4004-a009-42824f826ee8"}}]')
```

Comment:
why does the response data need to change here, your function doesn't actually do anything with the result so what value is there to having a different response other than making the code confusing. What you actually need to test here is that the responses were called with different url parameters than before when you pass in fields

Reply:
The purpose of the test is to verify the filtering fields. The URL path should be something like `api/workspaces?accessLevel,workspace.name,workspace.namespace,workspace.workspaceId`. But the `responses` seems matching the path before the path is combined with parameters. So, the URL is the same as the previous one. That's why I have to use `responses.replace()` instead of `responses.add()`.

Reply:
I probably change the function `list_anvil_workspaces()` to combine the parameters with the URL before calling the get method. It will be more friendly for the test.

Reply:
Could you replace this with your fixture data, like `WORKSPACE_RSP_ONE_VALID_PROJECT`? If not this is fine

Reply:
It is a different one. This one is used only for one time. So I keep it as a literal value here.